### ISSUE TITLE
Why don't handle source port and destination port?

### ISSUE THREAD

**Issue by fwqaaq** [zizifn/edgetunnel#284]
https://github.com/zizifn/edgetunnel/blob/4221293b0abd63cc139adb24f0e707548e488dfb/src/worker-vless.js#L540-L548

Complete UDP messge also includes source port and destination port(total 8 bytes), but it seems they are being ignored in here.

**Comment by zizifn** [comment]
Port and destination is in vless header. And worker  runtime not support udp yet.

**Comment by fwqaaq** [comment]
> Port and destination is in vless header. And worker runtime not support udp yet.

So vless sends UDP datas after processing, thus it also lacks a checksum?

**Comment by xqdoo00o** [comment]
I think in vless format, every udp chunk has a vless header, like tcp chunk.
so maybe it's not necessory to set udpStreamWrite.
and no need to use DOH, because some public dns like google 8.8.8.8 open a tcp port 53 to handle dns request, so `handleTCPOutBound` could handle dns.

**Comment by zizifn** [comment]
> I think in vless format, every udp chunk has a vless header, like tcp chunk. [source code](https://github.com/XTLS/Xray-core/blob/main/proxy/vless/encoding/encoding.go#L121). so maybe it's not necessory to set udpStreamWrite. and no need to use DOH, because some public dns like google 8.8.8.8 open a tcp port 53 to handle dns request, so `handleTCPOutBound` could handle dns.

You are right. In the init implementation, I used 8.8.8.8 for dns, but use cf doh for speed.
